internal/command/delta.go

```diff
@@ -70,26 +72,45 @@ func delta(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("preparing new deployment: %w", err)
 	}
 
-	// Create a new deployment delta
-	//
-	log.Print("Creating a new deployment delta...\n")
 	client, err := api.NewClient(apiUrl, apiToken, http.DefaultClient)
 	if err != nil {
 		return err
 	}
-	res, err := client.CreateDelta(context.Background(), orgID, appID, delta)
-	if err != nil {
-		return err
+
+	var res *ht.DeploymentDelta
+	if deltaUseId == "" {
+
+		log.Print("Creating a new deployment delta...\n")
+		res, err = client.CreateDelta(context.Background(), orgID, appID, delta)
+		if err != nil {
+			return err
+		}
+		res.Metadata.Url = fmt.Sprintf("%s/orgs/%s/apps/%s/envs/%s/draft/%s", uiUrl, orgID, appID, delta.Metadata.EnvID, res.ID)
+
+	} else {
+		log.Print("Retrieving existing delta...\n")
+		existing, err := client.GetDelta(context.Background(), orgID, appID, deltaUseId)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("Updating existing delta %s...\n", existing.ID)
+		updates := []*ht.UpdateDeploymentDeltaRequest{{Modules: delta.Modules}}
+		res, err = client.UpdateDelta(context.Background(), orgID, appID, deltaUseId, updates)
+		if err != nil {
+			return err
+		}
+		res.Metadata.Url = fmt.Sprintf("%s/orgs/%s/apps/%s/envs/%s/draft/%s", uiUrl, orgID, appID, delta.Metadata.EnvID, res.ID)
+
 	}
-	res.Metadata.Url = fmt.Sprintf("%s/orgs/%s/apps/%s/envs/%s/draft/%s", uiUrl, orgID, appID, delta.Metadata.EnvID, res.ID)
 
 	// Output resulting deployment delta
 	//
 	tmp, err := json.MarshalIndent(res, "", "  ")
 	if err != nil {
 		return err
 	}
-	os.Stdout.Write(tmp)
+	_, _ = os.Stdout.Write(tmp)
```

Comment:
Why do we need this extra assignment?

Reply:
Mostly just satisfying my IDE's strictness to not ignore errs 😆